Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/server/DccValidationServer.kt

```diff
@@ -120,11 +151,24 @@ class DccValidationServer @Inject constructor(
         }
     }
 
-    fun clear() {
-        Timber.d("clear()")
-        cache.evictAll()
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun getSource(response: Response<ResponseBody>) = if (response.notModified()) {
+        RuleSetSource.CACHE
+    } else {
+        RuleSetSource.SERVER
     }
 
+    data class RuleSetResult(
+        val ruleSetJson: String,
+        val source: RuleSetSource
+    )
+
+    enum class RuleSetSource {
+        SERVER, CACHE
+    }
+
+    private fun Response<ResponseBody>.notModified() = this.raw().networkResponse?.code == 304
```

Comment:
I'm using something similar in **CClConfigurationServer**. Maybe we should create an extension for this check and share it?

Reply:
Good point! Added it to `RetrofitExtensions`.